### PR TITLE
Add CSR common name to SANs if no other SANs are defined in CSR

### LINKED ISSUE #1171: [Bug]: CN not added as SAN
### Steps to Reproduce

1. Create CSR without specifying SAN:
```
openssl req \
  -new -newkey rsa:4096 \
  -out example.com.csr \
  -keyout example.com.pem \
  -subj "/C=AU/ST=Example/L=Example/O=Example Ltd./OU=Example Team/CN=example.com.crt"
```

2. Sign certificate:
```
step certificate sign \
  example.com.csr \
  /etc/step-ca/certs/root_ca.crt \
  /etc/step-ca/secrets/root_ca_key > example.com.crt
```

3. Check SAN of certificate:
```
openssl x509 -noout -ext subjectAltName -in example.com.crt
No extensions in certificate
```

### Your Environment

* OS - Ubuntu 22.04
* `step-ca` Version -
```
Smallstep CLI/0.25.1 (linux/amd64)
Release Date: 2023-11-29T03:45:04Z
```

### Expected Behavior

The common name should be automatically added as a SAN.

### Actual Behavior

The SAN is empty.

### Additional Context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction.
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by dopey**
- Fixes #1171

### CODE REVIEW

File: command/certificate/sign.go [hunk 1]
```diff
@@ -435,6 +446,19 @@ func createTemplateData(cr *x509.CertificateRequest, maxPathLen int) x509util.Te
...
+		for _, v := range sans {
+			if v == cr.Subject.CommonName {
+				cnInSANs = true
+				break
+			}
+		}
+		if !cnInSANs {
+			sans = append(sans, cr.Subject.CommonName)
+		}
+	}
```
Comment (maraino) [inline command/certificate/sign.go:460]: I think we want to do this only if there are no SANs.

File: command/certificate/sign.go [hunk 2]
```diff
@@ -174,6 +179,12 @@ $ step certificate sign \
 			flags.Template,
 			flags.TemplateSet,
 			flags.TemplateSetFile,
+			cli.BoolTFlag{
+				Name: "add-cn-san",
+				Usage: `Ensure that Common Name from CSR subject is added to Subject Alternative
+Name (SAN) extension of the certificate. This flag is enabled by default. To
+disable default behavior pass the flag as '--add-cn-san=false'.`,
+			},
```
Comment (maraino) [inline command/certificate/sign.go:187]: Let's do something like `--omit-cn-san` as @hslatman suggested.

**Review by maraino — CHANGES_REQUESTED** [review]

**Review by hslatman — DISMISSED** [review]
Nice 😄

File: command/certificate/sign.go [hunk 3]
```diff
@@ -435,6 +448,19 @@ func createTemplateData(cr *x509.CertificateRequest, maxPathLen int) x509util.Te
...
+		cnInSANs := false
+		for _, v := range sans {
+			if v == cr.Subject.CommonName {
+				cnInSANs = true
+				break
+			}
+		}
+		if !cnInSANs {
+			sans = append(sans, cr.Subject.CommonName)
+		}
```
Comment (maraino) [inline command/certificate/sign.go:461]: len(sans) is 0, so we don't need to do this. We can just do the following:

```suggestion
		sans = append(sans, cr.Subject.CommonName)
```

We might want to see what happens if the common name is empty and probably add a check for that.

**Review by maraino — CHANGES_REQUESTED** [review]

**Review by maraino — APPROVED** [review]

**Pull request merged**